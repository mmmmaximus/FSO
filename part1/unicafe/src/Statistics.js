import React from 'react'
import Stats from './Stats'

const Statistics = ({good, neutral, bad}) => {
  const all = good + neutral + bad
  const average = (good - bad) / all
  const positive = good / all * 100 + '%'

  if (all==0) {
   return <p>No feedback given</p> 
  }
  return (
    <div>
      <Stats text='good' number={good}/>
      <Stats text='neutral' number={neutral}/>
      <Stats text='bad' number={bad}/>
      <Stats text='all' number={all}/>
      <Stats text='average' number={average}/>
      <Stats text='positive' number={positive}/>
    </div>
  )
}

export default Statistics
