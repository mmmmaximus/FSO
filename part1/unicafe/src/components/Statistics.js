import React from 'react'
import Stats from './Stats'

const Statistics = ({good, neutral, bad}) => {
  const all = good + neutral + bad
  const average = (good - bad) / all
  const positive = good / all * 100 + '%'

  if (all===0) {
    return <p>No feedback given</p>
  }
  return (
    <table>
      <tbody data-testid='Statistics'>
        <Stats text='good' number={good}/>
        <Stats text='neutral' number={neutral}/>
        <Stats text='bad' number={bad}/>
        <Stats text='all' number={all}/>
        <Stats text='average' number={average}/>
        <Stats text='positive' number={positive}/>
      </tbody>
    </table>
  )
}

export default Statistics
