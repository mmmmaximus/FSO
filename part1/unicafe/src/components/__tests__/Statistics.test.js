import { render, screen } from '@testing-library/react'
import { within } from '@testing-library/dom'
import Statistics from './../Statistics'
import renderer from 'react-test-renderer'

test('component should render on screen', () => {
  // render Statistics component
  render(<Statistics good={1} neutral={2} bad={3}/>)
  const StatisticsElement = screen.getByTestId('Statistics')
  expect(StatisticsElement).toBeInTheDocument

  // render Stats component in Statistics
  const StatsInStatictics = within(StatisticsElement).getAllByTestId('Stats')
  expect(StatsInStatictics.length).toBe(6)
})

it('matches snapshot', () => {
  const tree = renderer.create(<Statistics good={1} neutral={2} bad={3}/>).toJSON()
  expect(tree).toMatchSnapshot()
})
