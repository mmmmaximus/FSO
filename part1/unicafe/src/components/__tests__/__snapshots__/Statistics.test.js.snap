// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<table>
  <tbody
    data-testid="Statistics"
  >
    <tr
      data-testid="Stats"
    >
      <td>
        good
      </td>
      <td>
        1
      </td>
    </tr>
    <tr
      data-testid="Stats"
    >
      <td>
        neutral
      </td>
      <td>
        2
      </td>
    </tr>
    <tr
      data-testid="Stats"
    >
      <td>
        bad
      </td>
      <td>
        3
      </td>
    </tr>
    <tr
      data-testid="Stats"
    >
      <td>
        all
      </td>
      <td>
        6
      </td>
    </tr>
    <tr
      data-testid="Stats"
    >
      <td>
        average
      </td>
      <td>
        -0.3333333333333333
      </td>
    </tr>
    <tr
      data-testid="Stats"
    >
      <td>
        positive
      </td>
      <td>
        16.666666666666664%
      </td>
    </tr>
  </tbody>
</table>
`;
